hack/patch-converter-runner/cmd/main.go

```diff
@@ -0,0 +1,198 @@
+)
+
+type config struct {
+	runtimePath    string
+	shootPath      string
+	kubeconfigPath string
+	outputPath     string
+}
+
+func main() {
+
+	var cfg config
+
+	flag.StringVar(&cfg.runtimePath, "runtime-path", "", "Path to the runtime CR file.")
+	flag.StringVar(&cfg.shootPath, "shoot-path", "", "Path to the shoot CR file.")
+	flag.StringVar(&cfg.kubeconfigPath, "kubeconfig-path", "", "Path to the kubeconfig file.")
+	flag.StringVar(&cfg.outputPath, "output-path", "", "Path to the resulting shoot CR file.")
+	flag.Parse()
+
+	printConfig(cfg)
+	inputRuntime, err := readRuntimeCRFromFile(cfg.runtimePath)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to read runtime CR: %v", err))
+	}
+
+	existingShoot, err := readShootFromFile(cfg.shootPath)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to read shoot CR: %v", err))
+	}
+
+	kcpClient, err := createKcpClient(cfg.kubeconfigPath)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to create Kubernetes client: %v", err))
+	}
+
+	converterConfig, err := getConverterConfig(kcpClient)
+	if err != nil {
+		panic(err)
+	}
+
+	patchConverter := converter.NewConverterPatch(converter.PatchOpts{
+		ConverterConfig:      converterConfig,
+		Workers:              existingShoot.Spec.Provider.Workers,
+		ShootK8SVersion:      existingShoot.Spec.Kubernetes.Version,
+		Extensions:           existingShoot.Spec.Extensions,
+		Resources:            existingShoot.Spec.Resources,
+		InfrastructureConfig: existingShoot.Spec.Provider.InfrastructureConfig,
+		ControlPlaneConfig:   existingShoot.Spec.Provider.ControlPlaneConfig,
+	})
+
+	updatedShoot, err := patchConverter.ToShoot(*inputRuntime)
+	if err != nil {
+		panic(err)
+	}
+
+	err = saveOutputShootFile(cfg.outputPath, &updatedShoot)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("Shoot CR generated successfully and saved to %s\n", cfg.outputPath)
+}
+
+func readRuntimeCRFromFile(filePath string) (*imv1.Runtime, error) {
+	return readObjectFromFile[imv1.Runtime](filePath)
+}
+
+func readShootFromFile(filePath string) (*gardener.Shoot, error) {
+	return readObjectFromFile[gardener.Shoot](filePath)
+}
+
+func readObjectFromFile[T any](filePath string) (*T, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	fileBytes, err := io.ReadAll(file)
+	if err != nil {
+		return nil, err
+	}
+
+	var obj T
+	err = yaml.Unmarshal(fileBytes, &obj)
+	if err != nil {
+		return nil, err
+	}
+
+	return &obj, nil
+}
+
+func createKcpClient(kcpKubeconfigPath string) (client.Client, error) {
+	restCfg, err := clientcmd.BuildConfigFromFlags("", kcpKubeconfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch rest config: %w", err)
+	}
+
+	scheme := runtime.NewScheme()
+	if err := addToScheme(scheme); err != nil {
+		return nil, err
+	}
+
+	return client.New(restCfg, client.Options{
+		Scheme: scheme,
+	})
+}
+
+func addToScheme(s *runtime.Scheme) error {
+	for _, add := range []func(s *runtime.Scheme) error{
```

Comment:
It will work for sure but here is small proposal to simplify 

```golang
if err := corev1.AddToScheme(s); err != nil {
    return fmt.Errorf("unable to add scheme: %w", err)
}

if err := v1.AddToScheme; err != nil {
    return fmt.Errorf("unable to add scheme: %w", err)
}

return nil
```

BTW Intellij reports type casting error in the first line, but it can be just IDE problem, because `go vet` tool passes fine
`
Cannot use 's' (type *"k8s. io/ apimachinery/ pkg/ runtime".Scheme) as the type *"github. com/ kyma-project/ infrastructure-manager/ vendor/ k8s. io/ apimachinery/ pkg/ runtime".Scheme `